<?xml version="1.0" ?>
<!DOCTYPE PISI SYSTEM "http://www.pardus.org.tr/projeler/pisi/pisi-spec.dtd">
<PISI>
    <Source>
        <Name>urlgrabber</Name>
        <Homepage>http://linux.duke.edu/projects/urlgrabber/</Homepage>
        <Packager>
            <Name>Faik Uygur</Name>
            <Email>faik@pardus.org.tr</Email>
        </Packager>
        <License>LGPLv2.1</License>
        <IsA>library</IsA>
        <Summary>High level cross-protocol url-grabber library for Python</Summary>
        <Description>Urlgrabber provides a clean library interface to protocol-independant file-access for Python programs</Description>
        <Archive sha1sum="866c5907a9ef072bea71d020079b4c53e44e1318" type="targz">http://linux.duke.edu/projects/urlgrabber/download/urlgrabber-3.1.0.tar.gz</Archive>
        <BuildDependencies>
            <Dependency>python</Dependency>
        </BuildDependencies>
        <Patches>
            <!-- Fixes "exceptions.AttributeError: SSLFactory instance has no attribute 'get_https_connection'" exception -->
            <Patch level="1">urlgrabber.ssl.patch</Patch>

            <Patch level="1">urlgrabber-keepalive.patch</Patch>
            <Patch level="1">urlgrabber-string-type.patch</Patch>
            <Patch level="1">urlgrabber-3.0.0-cleanup.patch</Patch>
            <Patch level="1">urlgrabber-ftp-port.patch</Patch>
            <Patch level="1">urlgrabber-progress-ui.patch</Patch>
            <!--<Patch level="1">urlgrabber-grab-no-range.patch</Patch>-->
            <Patch level="1">urlgrabber-no-ssl-ok.patch</Patch>
            <Patch level="1">urlgrabber-extra-progress.patch</Patch>
            <Patch level="1">urlgrabber-file-checkfunc.patch</Patch>
            <Patch level="1">md5-hashlib.patch</Patch>
            <Patch level="1">urlgrabber-3.0.0-progress-C-c+serial-console.patch</Patch>

            <!-- Fix range error -->
            <Patch level="1">fix-range-error.patch</Patch>
        </Patches>
    </Source>

    <Package>
        <Name>urlgrabber</Name>
        <RuntimeDependencies>
            <Dependency>python</Dependency>
        </RuntimeDependencies>
        <Files>
            <Path fileType="library">/usr/lib</Path>
            <Path fileType="doc">/usr/share/doc</Path>
        </Files>
    </Package>

    <History>
        <Update release="4">
            <Date>2009-06-13</Date>
            <Version>3.1.0</Version>
            <Comment>Fix a Range error which causes pisi to fail while trying to resume *.part files on some servers. Also disable fedora range patch which is bloated.</Comment>
            <Name>Ozan Çağlayan</Name>
            <Email>ozan@pardus.org.tr</Email>
        </Update>
        <Update release="3">
            <Date>2009-06-11</Date>
            <Version>3.1.0</Version>
            <Comment>Grab some important bug fixes from fedora.</Comment>
            <Name>Ozan Çağlayan</Name>
            <Email>ozan@pardus.org.tr</Email>
        </Update>
        <Update release="2">
            <Date>2009-02-11</Date>
            <Version>3.1.0</Version>
            <Comment>Removed unnecessary rename in install code</Comment>
            <Name>Bahadır Kandemir</Name>
            <Email>bahadir@pardus.org.tr</Email>
        </Update>
        <Update release="1">
            <Date>2007-11-21</Date>
            <Version>3.1.0</Version>
            <Comment>First release.</Comment>
            <Name>A. Murat Eren</Name>
            <Email>meren@pardus.org.tr</Email>
        </Update>
    </History>
</PISI>
